packages/TEMPLATE/install

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# To use if root is required
+[ "$UID" -eq 0 ] || (echo "installation requires root access"; exec sudo "$0" "$@")
+
+# Install general linux dependencies
+
+# General setup / configuration
+
+
+# Recipe for singularity images
+## Fetch the image
+singularity pull --name <image name> <singularity shub url>
+
+## Move it to the right place
+mv <image name> ~/.planutils/bin/images/
+
+
+# No need to move the "run" script -- if it exists, this will be handled for you
```

Comment:
Because the 3/4 files per package are in a custom python-ish location. You want the "run" script to be moved to the planutils bin directory and renamed to the shortname (so it's accessible via command line).

Reply:
Ahh, I see. So this would be something that happens when you install the package? I thought you wanted placeholder scripts that can respond with something like "this package is not yet installed, should we install it now?". If you have these placeholder scripts, they could do the job like this:

```bash
#!/bin/bash
planutils run $(basename "$0") $@
```

where `planutils run x $@` does (in python) something like this

```python
if not is_package_installed(x):
   ask_user_to_install(x)
   if not installed:
     sys.exit(0)
subprocess.exec(get_run_script(x), params)
```

both the placeholder script and the run script would always remain where they are.

Reply:
By the way, because I had this model in mind, I also suggested to "install" everything locally in a package-specific directory. Without this setup that suggestion doesn't make too much sense.

Reply:
Ah, I think I see what you mean. I think I'd keep the redirect at the bash level to avoid headaches with `argparse`. Single script and avoiding moving things on install/uninstall makes sense.